src/main/java/cs2103/v15_1j/jimjim/ui/TaskEventRowFactory.java

```diff
@@ -0,0 +1,492 @@
+package cs2103.v15_1j.jimjim.ui;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import com.jfoenix.controls.JFXCheckBox;
+
+import cs2103.v15_1j.jimjim.model.DataLists;
+import cs2103.v15_1j.jimjim.model.DeadlineTask;
+import cs2103.v15_1j.jimjim.model.Event;
+import cs2103.v15_1j.jimjim.model.FloatingTask;
+import javafx.geometry.HPos;
+import javafx.scene.control.Label;
+import javafx.scene.layout.GridPane;
+import javafx.scene.text.TextAlignment;
+
+public class TaskEventRowFactory {
+
+	private DataLists masterList;
+	private DataLists displayList;
+	private GridPane pane;
+
+	private Integer rowNo;
+
+	private DateTimeFormatter dateFmt;
+	private DateTimeFormatter dateTimeFmt;
+
+	private final double ID_LABEL_WIDTH = 50.0;
+	private final double NAME_LABEL_WIDTH = 350.0;
+	private final double DATE_LABEL_WIDTH = 200.0;
+	private final int NO_OF_COLUMNS = 3;
+
+	public TaskEventRowFactory(DataLists masterList, DataLists displayList, GridPane pane){
+		this.masterList = masterList;
+		this.displayList = displayList;
+		this.pane = pane;
+		this.rowNo = new Integer(0);
+
+		dateFmt = DateTimeFormatter.ofPattern("dd MMM yyyy");
+		dateTimeFmt = DateTimeFormatter.ofPattern("dd MMM h:mm a");
+	}
+
+	public void showTodayTaskEvents(){
+		rowNo = new Integer(-1);
+		pane.getChildren().clear();
+
+		Label headerLabel = new Label("Today");
+		headerLabel.getStyleClass().add("header");
+		pane.add(headerLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+
+		int noOfTaskEvents = showTaskEventsOnDate(LocalDate.now());
+
+		Label tempEmptyLabel = new Label("");
+		pane.add(tempEmptyLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+
+		if (noOfTaskEvents == 0){
+			pane.getChildren().remove(tempEmptyLabel);
+
+			Label noTaskEventsLabel = new Label("No Tasks or Events today");
+			noTaskEventsLabel.getStyleClass().add("no-overdue-label");
+			pane.add(noTaskEventsLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+		}
+	}
+
+	public void showTaskEventsFromDate(LocalDate selectedDate){
+		pane.getChildren().clear();
+		rowNo = new Integer(-1);
```

Comment:
I'm strongly against the use of a global variable like this. Can you just make it an argument that is passed around among these methods?